examples/spring-boot-with-certmanager-example/README.md

```diff
@@ -0,0 +1,122 @@
+
+## Generate the Keystore
+
+In this section, we're going to generate the `keystore.p12` using the password `supersecret`.
+
+Let's start with the password first. We need to use a Secret resource with the value `supersecret` in the data key `password`:
+
+```
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pkcs12-pass
+data:
+  # "supersecret" in base64:
+  password: c3VwZXJzZWNyZXQ=
+type: Opaque
+```
+
+In Dekorate, we can include these resources in the file `src/main/resources/k8s/common.yml`. To include these resources as part of the final kubernetes.yml file, we need to configure Dekorate as follows:
+
+```
+dekorate.options.input-path=k8s
+```
+
+Let's now configure the certificate to generate the previous Keystore:
+
+```
+dekorate.certificate.secret-name=tls-secret
+dekorate.certificate.self-signed.enabled=true
+dekorate.certificate.keystores.pkcs12.create=true
+# the secret name of the password (see above):
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.name=pkcs12-pass 
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.key=password
+```
+
+Using this configuration, Cert-Manager will generate a self-signed certificate and the Keystore files within the secret `tls-secret`. Therefore, we need to map the data within this secret `tls-secret` into the Spring Boot properties `server.ssl.key-store` and `server.ssl.key-store-password`.
+
+Moreover, the Cert-Manager Dekorate extension will configure the application resources to mount a volume with the generated certificates at the path `/etc/certs` (it can be configured using `dekorate.certificate.volume-mount-path`).
+
+So, the final dekorate properties that we need to add are the following:
+
+```
+dekorate.kubernetes.env-vars[0].name=SERVER_SSL_KEY_STORE
+dekorate.kubernetes.env-vars[0].value=/etc/certs/keystore.p12
+dekorate.kubernetes.env-vars[1].name=SERVER_SSL_KEY_STORE_PASSWORD
+dekorate.kubernetes.env-vars[1].secret=pkcs12-pass
+dekorate.kubernetes.env-vars[1].value=password
+```
+
+## Run the application in Kubernetes
+
+First, make sure you have connected/logged to a kubernetes cluster and installed [Cert-Manager](https://cert-manager.io/docs/installation/) in the cluster.
+
+Next, we need to generate the manifests and push the application container image to our container registry:
+
+```
+mvn clean install -Ddekorate.push=true -Ddekorate.docker.registry=<your container registry. Example: quay.io> -Ddekorate.docker.group=<the container group. Example: user>
+```
+
+After running the above command, we will have the generated manifests in the file `target/classes/META-INF/dekorate/kubernetes.yml`. So, let's install it:
+
+```
+kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml
+```
+
+Now, we should see the secret resource named `tls-secret` that `Cert-Manager` created with the expected keystore file:
```

Comment:
"Now, we should see the secret resource named `tls-secret` that `Cert-Manager` created with the expected keystore file:" -->

"After a few moment, we should be able to see the secret resource named `tls-secret` that the `Cert-Manager` has created like the pkcs12 keystore file: